pipeline/authn/authenticator_oauth2_introspection_test.go

```diff
@@ -840,4 +840,22 @@ func TestAuthenticatorOAuth2Introspection(t *testing.T) {
 			require.NotEqual(t, noPreauthClient3, noPreauthClient)
 		})
 	})
+
+	t.Run("unmarshal-audience", func(t *testing.T) {
+		t.Run("Should pass because audience is a valid string", func(t *testing.T) {
+			var r AuthenticatorOAuth2IntrospectionResult
+			data := `{"aud":"audience"}`
+			json.Unmarshal([]byte(data), &r)
+			require.NoError(t, err)
+			require.Equal(t, Audience{"audience"}, r.Audience)
```

Comment:
Wouldn't it make more sense to test the `Audience` type instead?
```suggestion
			var r Audience
			data := `"audience"`
			json.Unmarshal([]byte(data), &r)
			require.NoError(t, err)
			require.Equal(t, Audience{"audience"}, r)
```

Reply:
I think both approaches are fine. The current test ensures that the outer struct also correctly decodes, which is what we want here!

Reply:
@zepatrik Would be easier, yes. I changed it. @aeneasr I don't think it makes any difference in this case. As the outer struct only containes the audience, it adds no value.